torch_xla/csrc/init_python_bindings.cpp

```diff
@@ -1437,7 +1437,8 @@ void InitXlaModuleBindings(py::module m) {
       // with a different sharding then we block & ask the user to clear
       // the existing sharding first.
       auto current_sharding_spec = xtensor->sharding_spec();
-      if (current_sharding_spec) {
+      if (current_sharding_spec && (current_sharding_spec->sharding.type() !=
```

Comment:
More of a general comment, but what do you think about adding a TraceMe for the profiler after this check (at L1448)? That way it's easy to tell if a TransferFromServer call is due to resharding.

Reply:
Good idea :)